internal/runtime/client/client_test.go

```diff
@@ -656,8 +740,144 @@ func TestClient_CallExtension(t *testing.T) {
 	}
 }
 
+func Test_aggregateResponses(t *testing.T) {
+	tests := []struct {
+		name              string
+		aggregateResponse runtimehooksv1.ResponseObject
+		responses         []runtimehooksv1.ResponseObject
+		want              runtimehooksv1.ResponseObject
+	}{
+		{
+			name:              "Aggregate response if there is only one response",
+			aggregateResponse: fakeSuccessResponse(),
+			responses: []runtimehooksv1.ResponseObject{
+				fakeSuccessResponse(),
+			},
+			want: fakeSuccessResponse(),
+		},
+		{
+			name:              "Aggregate retry response if there is only one response",
+			aggregateResponse: fakeRetryableSuccessResponse(0),
+			responses: []runtimehooksv1.ResponseObject{
+				fakeRetryableSuccessResponse(5),
+			},
+			want: fakeRetryableSuccessResponse(5),
+		},
+		{
+			name:              "Aggregate retry responses to lowest non-zero retryAfterSeconds value",
+			aggregateResponse: fakeRetryableSuccessResponse(0),
+			responses: []runtimehooksv1.ResponseObject{
+				fakeRetryableSuccessResponse(0),
+				fakeRetryableSuccessResponse(1),
+				fakeRetryableSuccessResponse(5),
+				fakeRetryableSuccessResponse(4),
+				fakeRetryableSuccessResponse(3),
+			},
+			want: fakeRetryableSuccessResponse(1),
```

Comment:
Q: should this be 0 or am I wrong?

Reply:
Maybe I'm wrong, but I think our goal was to aggregate to the lowest **non-zero** value. (I think essentially we want to use the lowest RetryAfterSeconds except 0, because 0 is the same as no retry)